server/monitor-types/tailscale-ping.js

```diff
@@ -0,0 +1,90 @@
+     */
+    async check(monitor, heartbeat) {
+        try {
+            let tailscaleOutput = await this.runTailscalePing(monitor.url);
+            this.parseTailscaleOutput(tailscaleOutput, heartbeat);
+        } catch (err) {
+            // trigger log function somewhere to display a notification or alert to the user (but how?)
+            throw new Error(`Error checking Tailscale ping: ${err}`);
+        }
+    }
+
+    /**
+     * Runs the Tailscale ping command to the given URL.
+     * 
+     * @param {string} url - The URL to ping.
+     * @returns {Promise<string>} - A Promise that resolves to the output of the Tailscale ping command
+     * @throws Will throw an error if the command execution encounters any error.
+     */
+    async runTailscalePing(url) {
+        let cmd = `tailscale ping ${url}`;
+        return new Promise((resolve, reject) => {
+            let timeout = this.interval * 1000 * 0.8;
+            exec(cmd, {timeout: timeout}, (error, stdout, stderr) => {
+                // we may need to handle more cases if tailscale reports an error that isn't necessarily an error (such as not-logged in or DERP health-related issues) 
+                if (error) {
+                    reject(`Execution error: ${error.message}`);
+                    return;
+                }
+                if (stderr) {
+                    reject(`Error in output: ${stderr}`);
+                    return;
+                }
+                resolve(stdout);
+            });
+        });
+    }
+
+    /**
+     * Parses the output of the Tailscale ping command to update the heartbeat.
+     * 
+     * @param {string} tailscaleOutput - The output of the Tailscale ping command.
+     * @param {Object} heartbeat - The heartbeat object to update.
+     * @throws Will throw an eror if the output contains any unexpected string.
+     */
+    parseTailscaleOutput(tailscaleOutput, heartbeat) {
+        let lines = tailscaleOutput.split('\n');
+
+        for (let line of lines) {
+            if (line.includes("pong from")) {
+                heartbeat.status = UP;
+                let time = line.split(" in ")[1].split(" ")[0];
+                heartbeat.ping = parseInt(time);
+                heartbeat.msg = line;
+                break;
+            } else if (line.includes("timed out")) {
+                throw new Error(`Ping timed out: "${line}"`);
+                // Immediately throws upon "timed out" message, the server is expected to re-call the check function
+            } else if (line.includes("no matching peer")) {
+                throw new Error(`Nonexistant or inaccessible due to ACLs: "${line}"`);
+            } else if (line.includes("is local Tailscale IP")) {
+                throw new Error(`Tailscale only works if used on other machines: "${line}"`);
+            } else if (line !== '') {
+                throw new Error(`Unexpected output: "${line}"`);
+            }
```

Comment:
I think it could not pass the eslint.

Reply:
I adjusted the suggestion to be linting-compliant.

Reply:
> I adjusted the suggestion to be linting-compliant. (Now I am unshure if this is a better suggestion)

Since the changes made would be minimal, and that the code is written to show that each condition is mutually exclusive, although the new revision _could_ be lint-complient, I believe that the current structure better communicates the logic of this codeblock.
Would you agree? If yes, react to this comment and I'll mark it resolved, otherwise, please let me know if there are other reasons to keep this change.